src/domain/usecases/CreateIssueUseCase.ts

```diff
@@ -0,0 +1,109 @@
+import { FutureData } from "$/data/api-futures";
+import { IssueRepository } from "$/domain/repositories/IssueRepository";
+import { UCIssue } from "$/domain/usecases/common/UCIssue";
+import { QualityAnalysisIssue } from "$/domain/entities/QualityAnalysisIssue";
+import { Id, Period } from "$/domain/entities/Ref";
+import { Future } from "$/domain/entities/generic/Future";
+import { UCAnalysis } from "$/domain/usecases/common/UCAnalysis";
+import { QualityAnalysisRepository } from "$/domain/repositories/QualityAnalysisRepository";
+import { QualityAnalysis } from "$/domain/entities/QualityAnalysis";
+import { Maybe } from "$/utils/ts-utils";
+
+export type IssueTemplate = {
+    categoryOptionComboId: Maybe<Id>;
+    countryId: Maybe<Id>;
+    dataElementId: Maybe<Id>;
+    description: string;
+    period: Maybe<Period>;
+};
+
+export class CreateIssueUseCase {
+    private issueUseCase: UCIssue;
+    private analysisUseCase: UCAnalysis;
+
+    constructor(
+        private analysisRepository: QualityAnalysisRepository,
+        private issueRepository: IssueRepository
+    ) {
+        this.issueUseCase = new UCIssue(this.issueRepository);
+        this.analysisUseCase = new UCAnalysis(this.analysisRepository);
+    }
+
+    execute(options: CreateIssueUseCaseOptions): FutureData<Maybe<QualityAnalysis>> {
+        const { qualityAnalysisId, sectionId, issues } = options;
+
+        if (issues.length <= 0) return Future.success(undefined);
```

Comment:
Not need to change, just a mention we have a `Future.void()` method which is equivalent to `Futures.success(undefined)`